### PR TITLE
Paginate events

### PR DESCRIPTION
**Pull request by bachstatter**
Added a new getEvents function that looks at `fromBlock` and `toBock` and splits things up as needed.
I've set `MAX_BLOCK` to 2000, do you think that's reasonable?

### CODE REVIEW

File: src/keeper-helpers.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { ethers } from "ethers";
+import { range } from "lodash";
+
+const MAX_BLOCKS = 2000;
+// exported for test
+export const getPaginatedFromAndTo = (fromBlock: number, toBlock: number) => {
+  const numberOfBlocks = toBlock - fromBlock;
+  if (numberOfBlocks > MAX_BLOCKS) {
```
Comment (artdgn) [inline src/keeper-helpers.ts:8]: nit: ideally, shouldn't be needed and should return the correct result if numberOfBlocks <= MAX_BLOCKS

**Review by artdgn — CHANGES_REQUESTED** [review]
We probably shouldn't start from first block, since most of the history of the chain will have no events, but we will still makes those requests.

I didn't find an easy way to get the contract creation block / tx. So probably we can get by by just having this as hard coded value (for the initial deployment of futures).

File: src/keeper-helpers.ts (hunk 1 above)
Comment (bachstatter) [reply src/keeper-helpers.ts:8]: Hmm we would need some logic for when `numberOfBlocks` is 0, which it is when we're just processing 1 block. 
Can remove the if statement and do this:
```
const numberOfBlocks = toBlock - fromBlock || 1
```
If you think thats clearer?

**Comment by bachstatter** [comment]
> We probably shouldn't start from first block, since most of the history of the chain will have no events, but we will still makes those requests.
> 
> I didn't find an easy way to get the contract creation block / tx. So probably we can get by by just having this as hard coded value (for the initial deployment of futures).

Yea make sense. Do you know what block number we should use?
Can probably put it as default here: 
https://github.com/Synthetixio/futures-keepers/blob/paginate-events/src/run/run.ts#L14

**Review by artdgn — APPROVED** [review]

**Pull request merged**